x-pack/plugins/files/server/saved_objects/file.ts

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { SavedObjectsType } from '@kbn/core/server';
+import { FILE_SO_TYPE } from '../../common';
+import type { FileSavedObjectAttributes } from '../../common';
+
+export const fileObjectType: SavedObjectsType<FileSavedObjectAttributes> = {
+  name: FILE_SO_TYPE,
+  hidden: true,
+  namespaceType: 'multiple-isolated',
+  mappings: {
+    dynamic: false,
+    properties: {
+      created_at: {
+        type: 'date',
+      },
+      updated_at: {
+        type: 'date',
+      },
+      name: {
+        type: 'keyword',
+      },
+      status: {
+        type: 'keyword',
+      },
+      created_by: {
+        type: 'keyword',
+      },
+      storage_id: {
+        type: 'keyword',
+      },
+      content_ref: {
+        type: 'keyword',
+      },
+      content_type: {
```

Comment:
I was thinking we could split the mime-type into two fields: `mime` and `extension`. The first part of mime-type would be in `mime` fields and the latter in `extension`. For example say your mime-type is `image/png`, we would have:

```
{
  mime: 'image',
  extension: 'png',
}
```

This way we could those fields to filter more easily.

- Say we want all images, we filter by the `mime` field.
- Say we want all files with a given extension, we filter by the `extension` field.